vignettes/generic.Rmd

```diff
@@ -0,0 +1,385 @@
+---
+title: "Generic Functions"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Generic Functions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(admiraldev)
+```
+
+# Introduction
+
+This vignette explains some of the `{admiral}` generic functions, which we believe
+will most commonly be able to be re-used to save users having to create their own
+functions for study-specific analysis derivations. These re-usable functions give
+the users great flexibility, but the challenge is knowing which to use when. So
+this vignette is intended to help understand the differences between each, thus
+improving the findability of the function needed for each task.
+
+The focus will be on the following different sets of generic functions:
+
+* [_merged_ functions](#merged)
+* [_joined_ functions](#joined)
+* [_extreme_ functions](#extreme)
+
+In each section below, we will show examples of different functions that belong
+to each of these sets, but we don't list out each and every function offered.
+Users should refer to the Reference page and search for example "merged" to see
```

Comment:
I think it would be nice if we had a dedicated section for the merged, joined, and extreme functions in the reference (and have links here pointing to them). For example `derive_var_confirmation_flag()` belongs to the "joined" functions but you won't find it if you search for "joined".

It is easy to add such sections to the reference page but it would result in listing some functions more than once.

Reply:
i was thinking of this as we do call out a similar list in the date imputation vignette, but then i was thinking it would take some maintenance and we'd have to remember to add here each time we added a new function. i'm going to add a discussion topic to Wednesday's call on this vignette and we can see what people think

Reply:
Let's leave this conversation open and come back to it for https://github.com/pharmaverse/admiral/issues/1522 later.